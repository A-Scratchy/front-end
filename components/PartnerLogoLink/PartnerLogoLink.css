.PartnerLogoLink {
  filter: grayscale(100%);
  display: flex;
  flex-basis: 200px;
  justify-content: center;
}

.PartnerLogoLink:hover {
  filter: grayscale(0);
}

.logo {
  height: 100px;
}
