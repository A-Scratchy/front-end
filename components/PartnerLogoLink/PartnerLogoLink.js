import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { s3 } from 'common/constants/urls';
import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
import styles from './PartnerLogoLink.css';

export default class PartnerLogoLink extends Component {
  static propTypes = {
    className: PropTypes.string,
    logo: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  };

  static defaultProps = {
    className: '',
  };

  render() {
    const { className, logo, name, url } = this.props;

    return (
      <div className={classNames(className, styles.PartnerLogoLink)}>
        <OutboundLink
          ref={url}
          analyticsEventLabel={`Opening partner link ${name}`}
          hasIcon={false}
        >
          <img className={styles.logo} src={`${s3}${logo}`} alt={name} />
        </OutboundLink>
      </div>
    );
  }
}
