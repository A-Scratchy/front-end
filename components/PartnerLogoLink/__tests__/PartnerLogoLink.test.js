/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import PartnerLogoLink from '../PartnerLogoLink';

describe('PartnerLogoLink', () => {
  it('should render with required props', () => {
    createSnapshotTest(<PartnerLogoLink>Test</PartnerLogoLink>);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(<PartnerLogoLink className="test-class">Test</PartnerLogoLink>);
  });
});
