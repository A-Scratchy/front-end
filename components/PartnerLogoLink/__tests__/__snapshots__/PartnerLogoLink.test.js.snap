// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnerLogoLink should render with many props assigned 1`] = `
<div
  className="test-class PartnerLogoLink"
>
  <a
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="screenReaderOnly"
    >
      Opens in new window
    </span>
    <img
      className="logo"
      src="https://s3.amazonaws.com/operationcode-assets/undefined"
    />
  </a>
</div>
`;

exports[`PartnerLogoLink should render with required props 1`] = `
<div
  className="PartnerLogoLink"
>
  <a
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="screenReaderOnly"
    >
      Opens in new window
    </span>
    <img
      className="logo"
      src="https://s3.amazonaws.com/operationcode-assets/undefined"
    />
  </a>
</div>
`;
